earthaccess/search.py

```diff
@@ -851,6 +851,26 @@ def point(self, lon: FloatLike, lat: FloatLike) -> Self:
         """
         return super().point(lon, lat)
 
+    def multipoint(self, lon_lat_pairs: Sequence[PointLike]) -> Self:
+        """Filter by granules that include multiple geographic points.
+
+        Parameters:
+            lon_lat_pairs: sequence of (lon, lat) tuples
+
+        Returns:
+            self
+        """
+        points = []
+
+        for x, y in lon_lat_pairs:
+            self.point(x, y)
+            points.append(self.params.pop('point')[0])
+        
+        self.params['point'] = points
+        self.options['point'] = {'or': True}        
```

Comment:
I would prefer not to assume the caller wants the `or` option to be set to `True`, so let's add an `or_` kwarg, defaulted to `None`:

```suggestion
        if or_ is not None:
            self.options['point'] = {'or': or_}
```